Compilation of generated code without some replacement of input parameters

/go/src/sample/rules.go:794:176: syntax error: unexpected customerid at end of statement

GENERATED CODE:
output["hide"] = "[map[screenId:transferscreen _condition:contains(group:showTransfersScreen,input:customerid) _conditionCode:contains(group:showTransfersScreen,ctx.Input["customerid"].(string))] map[screenId:accountscreen _condition:before('2020-04-30T23:32:21+00:00') _conditionCode:before("2020-04-30T23:32:21+00:00")] map[_condition:contains(group:specialCustomers,input:customerid) and randomPerc(10, input:customerid) _conditionCode:contains(group:specialCustomers,ctx.Input["customerid"].(string)) && randomPerc(10,ctx.Input["customerid"].(string),1234) screenId:specialcustomersscreen]]"

--- a/ruller-dsl-feature-flag/main.go
+++ b/ruller-dsl-feature-flag/main.go
@@ -364,7 +364,7 @@ func conditionCode(value interface{}, inputTypes map[string]ruller.InputType, ru
 		//_condition="input:age > 30 and input:name='stutz'" ---> "input:age.(float64) > 30 and input:name.(string)=='stutz'"
 
 		//find all numeric comparisons
-		numberInputRegex := regexp.MustCompile("input:([a-z0-9-_]+)\\s*[><==]\\s*[0-9]+")
+		numberInputRegex := regexp.MustCompile("input:([a-z0-9-_]+)\\s*([><]|==|!=)\\s*[0-9]+")
 		numberMatches := numberInputRegex.FindAllStringSubmatch(condition, -1)
 		for _, numberMatch := range numberMatches {
 			logrus.Debugf("Condition number match %s - %s", numberMatch[0], numberMatch[1])

--- a/sample-json/menu.json
+++ b/sample-json/menu.json
@@ -58,6 +58,26 @@
                     "_condition": "input:_ip_state~='DF|RJ'"
                 }
             ]
+        },
+        {
+            "label": "menu3",
+            "_condition": "input:age == 44",
+            "_items": [
+                {
+                    "label": "menu3.1",
+                    "uri": "/menu3.1"
+                }
+            ]
+        },
+        {
+            "label": "menu4",
+            "_condition": "input:age != 44",
+            "_items": [
+                {
+                    "label": "menu4.1",
+                    "uri": "/menu4.1"
+                }
+            ]
         }
     ]
 }

